src/coinjoin/coinjoin.cpp

```diff
@@ -306,6 +306,25 @@ bool CCoinJoinBaseSession::IsValidInOuts(const CTxMemPool& mempool, const std::v
     return true;
 }
 
+// Responsibility for checking fee sanity is moved from the mempool to the client (BroadcastTransaction)
+// but CoinJoin still requires ATMP with fee sanity checks so we need to implement them separately
```

Comment:
CoinJoin code expects that ATMP will check fee sanity because unlike every other non-RPC `AcceptToMemoryPool` call which uses an `nAbsurdFee` value of `0` (to skip absurd fee checking), CoinJoin code in a few places uses `DEFAULT_MAX_RAW_TX_FEE` (not to be confused with `DEFAULT_MAX_RAW_TX_FEE_RATE`, which has the same value, except as `CFeeRate` as opposed to `CAmount`) as `nAbsurdFee`, triggering the fee check.

Perhaps it could be refactored to be removed entirely but that's outside the scope of the PR and the backport attempts to preserve functionality as-is on the CoinJoin front.